### PR TITLE
Small appcompat fixes

### PR DESCRIPTION
**Pull request by learn-more**
Some local changes that I wanna run on the builders :)

### CODE REVIEW

File: dll/appcompat/apphelp/sdbapi.c [hunk 1]
```diff
@@ -177,15 +177,23 @@ BOOL WINAPI SdbpOpenMemMappedFile(LPCWSTR path, PMEMMAPPED mapping)
...
-    }
+    RtlInitUnicodeString(&FileName, path);
 
     InitializeObjectAttributes(&ObjectAttributes, &FileName, OBJ_CASE_INSENSITIVE, NULL, NULL);
     Status = NtOpenFile(&mapping->file, GENERIC_READ | SYNCHRONIZE, &ObjectAttributes, &IoStatusBlock, FILE_SHARE_READ, FILE_SYNCHRONOUS_IO_NONALERT);
-    RtlFreeUnicodeString(&FileName);
+
+    if (Status == STATUS_OBJECT_NAME_INVALID || Status == STATUS_OBJECT_PATH_SYNTAX_BAD)
+    {
+        if(!RtlDosPathNameToNtPathName_U(path, &FileName, NULL, NULL))
```
Comment (HBelusca) [inline dll/appcompat/apphelp/sdbapi.c]: "if"<space>"(etc...)"

File: dll/appcompat/apphelp/sdbapi.c [hunk 2]
```diff
@@ -177,15 +177,23 @@ BOOL WINAPI SdbpOpenMemMappedFile(LPCWSTR path, PMEMMAPPED mapping)
...
+        {
+            RtlFreeUnicodeString(&FileName);
```
Comment (ThFabba) [inline dll/appcompat/apphelp/sdbapi.c]: This is odd in a failure case. What are you freeing here? Something the function put in halfway through? `path`?

Comment (learn-more) [reply dll/appcompat/apphelp/sdbapi.c]: This free should indeed be removed, thanks.

File: dll/appcompat/apphelp/sdbapi.c (hunk 1 above)
Comment (learn-more) [reply dll/appcompat/apphelp/sdbapi.c]: Yeah I copied it from above (see the removed block).
Will fix.

**Pull request merged**